src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -581,6 +581,16 @@ private static void BlockAssertsWithoutExperimentalFeatures(SemanticModel model,
                     diagnostics.Write(assert.DeclaringAssert, x => x.AssertsUnsupported());
                 }
             }
+            foreach (var resourceDeclarationSymbol in model.Root.ResourceDeclarations)
+            {
+                if (resourceDeclarationSymbol.TryGetBodyProperty(LanguageConstants.ResourceAssertPropertyName)?.Value != null)
+                {
+                    if (!model.Features.AssertsEnabled)
+                    {
+                        diagnostics.Write(resourceDeclarationSymbol.TryGetBodyProperty(LanguageConstants.ResourceAssertPropertyName)!.Value, x => x.AssertsUnsupported());
+                    }
+                }
+            }
```

Comment:
You may use pattern matching to avoid calling `TryGetBodyProperty` twice and using the non-null assertion operator:
```suggestion
                if (resourceDeclarationSymbol.TryGetBodyProperty(LanguageConstants.ResourceAssertPropertyName) is { Value: var value } &&
                    !module.Features.AssertsEnabled)
                {
                    diagnostics.Write(value, x => x.AssertsUnsupported());
                }
            }
```

Reply:
Thank you for the great suggestion! This is so cool, much shorter 😆